### PR TITLE
CI/TST: Run http server in process instead of thread for test_user_agent.py

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them

Part 3 in trying to avoid timeouts in this test

### CODE REVIEW

File: pandas/tests/io/test_user_agent.py [hunk 1]
```diff
@@ -181,6 +184,47 @@ def do_GET(self):
...
+
+
+def process_server(responder, port):
+    with http.server.HTTPServer(("localhost", port), responder) as server:
+        server.handle_request()
+    server.server_close()
+
+
+@pytest.fixture
+def responder(request):
```
Comment (jbrockmendel) [inline pandas/tests/io/test_user_agent.py:205]: can you comment on why all this is necessary

Comment (mroeschke) [reply pandas/tests/io/test_user_agent.py:206]: Added some comments

**Review by jbrockmendel — APPROVED** [review]
LGTM; CI failures are unrelated codecov

**Comment by jreback** [comment]
thanks @mroeschke

**Pull request merged**